src/main/java/edu/kis/powp/jobs2d/command/canvas/CanvasVisitor.java

```diff
@@ -0,0 +1,40 @@
+package edu.kis.powp.jobs2d.command.canvas;
+
+import edu.kis.powp.jobs2d.command.*;
+
+import java.util.Iterator;
+import java.util.logging.Logger;
+
+public class CanvasVisitor implements CommandVisitor {
+    private Canvas currentCanvas;
+    private final static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+    public CanvasVisitor(Canvas canvas) {
+        currentCanvas = canvas;
+    }
+
+    @Override
+    public void visit(OperateToCommand operateToCommand) {
+        if(!currentCanvas.isInsideCanvas(operateToCommand.getX(), operateToCommand.getY()))
+        {
+            logger.info("Point (" + operateToCommand.getX() + "," + operateToCommand.getY() + ") is exceeding canvas.");
```

Comment:
Do not implement an action within the visitor. Instead provide a getter to obtain the result of the visitation.